ui/components/users/profile/role-item.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import { Chip } from "@nextui-org/react";
+import { Ban, Check } from "lucide-react";
+import { useState } from "react";
+
+import { Role, RoleDetail } from "@/types/users/users";
+
+interface PermissionItemProps {
+  enabled: boolean;
+  label: string;
+}
+
+export const PermissionIcon = ({ enabled }: { enabled: boolean }) => (
+  <span
+    className={`inline-flex h-4 w-4 items-center justify-center rounded-full ${enabled ? "bg-green-100 text-green-700" : "bg-red-100 text-red-500"}`}
+  >
+    {enabled ? <Check /> : <Ban />}
+  </span>
+);
+
+const PermissionItem = ({ enabled, label }: PermissionItemProps) => (
+  <div className="flex items-center space-x-2 whitespace-nowrap">
+    <PermissionIcon enabled={enabled} />
+    <span className="text-xs">{label}</span>
+  </div>
+);
+
+export const RoleItem = ({
+  role,
+  roleDetail,
+}: {
+  role: Role;
+  roleDetail?: RoleDetail;
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  if (!roleDetail) {
+    return (
+      <Chip key={role.id} size="sm" variant="flat" color="primary">
+        {role.id}
+      </Chip>
+    );
+  }
+
+  const { attributes } = roleDetail;
+  const roleName = attributes?.name || role.id;
+  const permissionState = attributes?.permission_state || "";
+  const detailsId = `role-details-${role.id}`;
+
+  const permissions = [
```

Comment:
can we move this to a helper file? We can reuse it in both `add-role-form.tsx` and `edit-role-form.tsx`.

 No need to update those files in this PR, I'll create a follow-up ticket for that. Thanks!